### PR TITLE
Support if/else expressions in EEx

### LINKED ISSUE #33: Doesn't support else (?)
Had no luck with this:

``` slim
      = if @meta do
        input type="checkbox" name="meta" value="true" checked="checked"
      = else  # also tried "- else"
        input type="checkbox" name="meta" value="true"
```

So I ended up working around it like this:

``` slim
      = if @meta do
        input type="checkbox" name="meta" value="true" checked="checked"

      = unless @meta do
        input type="checkbox" name="meta" value="true"
```

### PR DESCRIPTION
**Pull request by doomspork**
RFC @Rakoth, @henrik —  not the most attractive fix but the simplest one I could think of (at the moment) for ensuring an `else` gets nested inside the appropriate `if`.

Thoughts?

This closes #33

### CODE REVIEW

File: lib/slim_fast/parser.ex [hunk 1]
```diff
@@ -234,7 +234,9 @@ defmodule SlimFast.Parser do
     trimmed  = String.lstrip(line)
     trim_len = String.length(trimmed)
 
-    {orig_len - trim_len, trimmed}
+    offset = if trimmed == "- else", do: 2, else: 0
```
Comment (doomspork) [inline lib/slim_fast/parser.ex:237]: Offset any `- else` so it gets nested under the `if`.

**Comment by henrik** [comment]
Cool, nice job! Could there be similar issues with other multi-branch things? Case, cond?

**Comment by doomspork** [comment]
@henrik that's a good question, do you have an example in mind?

**Comment by henrik** [comment]
@doomspork None I have tried, but I was thinking of constructs like

```
= case @something do
- "foo" ->
  p hello foo
- "bar" ->
  p hello bar
```

**Comment by doomspork** [comment]
@henrik no that would not work but this does:

``` slim
= case @something do
  - "foo" ->
    p hello foo
  - "bar" ->
    p hello bar
```

I'm not sure which is "correct", I certainly prefer the indenting.

**Comment by doomspork** [comment]
Any thoughts @Rakoth?

**Comment by stevegraham** [comment]
+1 on the case example indenting also

**Comment by henrik** [comment]
Oh yeah, that makes sense 👍 If the error message is obscure otherwise, maybe docing it would be worthwhile.

**Comment by Rakoth** [comment]
Indenting is great, but I am still getting an error message with it: `expected -> clauses for do in case`, so eex expect each clause on its own line, but we get it compiled into one line

**Comment by doomspork** [comment]
Good catch @Rakoth.   The easiest fix for that seems to be including a newline between the opening and ending tags in `render_branch/2`:

``` elixir
  defp render_branch(%{} = branch) do
    opening = branch.attributes
              |> Enum.map(fn {k, v} -> render_attribute(k, v) end)
              |> Enum.join(" ")
              |> open(branch)

    closing = close(branch)
    opening <> "\n" <> compile(branch.children) <> "\n" <> closing
  end
```

Does require a number of test changes to account for the new addition but the output evaluates.

This slim:

``` slim
= case something do
  - "foo" ->
    p hello foo
  - "bar" ->
    p hello bar
```

Compiles to:

``` erb
<%= case something do %>\n<% \"foo\" -> %><p>hello foo</p>\n<% \"bar\" -> %><p>hello bar</p>\n<% end %>"
```

Given `[something: "foo"]` it evaluates to:

``` html
"\n<p>\nhello foo\n</p>\n"
```

**Comment by doomspork** [comment]
On second thought a less impactful change would be adding the newline only to EEx opening tags as such:

``` elixir
  defp open(_, %{type: :eex, content: code, attributes: attrs}) do
    inline = if attrs[:inline], do: "=", else: ""
    "<%#{inline} #{code} %>\n"
  end
```

In this case the aforementioned slim example would evaluate to: `\n<p>hello foo</p>`

**Comment by doomspork** [comment]
I made the aforementioned change.  The additional `\n` aren't pretty but they didn't break the markup in the few examples I tried.  @Rakoth any thoughts?

**Comment by doomspork** [comment]
If there are no objections to these changes I'm going to merge this week and get this functionality out into the wild.

**Comment by stevegraham** [comment]
🙌

**Pull request merged**